### ISSUE TITLE
Add more operator options for rules with date fields in the Query Builder

### ISSUE THREAD

**Issue by robyngit** [NCEAS/metacatui#1571]
Currently, when building a query rule with one of the date fields, a user can only select "is between" as the operator. It would be nice to include options for "is before" (max only) "is after" (min only), and "is in the year" (value only). The behaviour of the date filter model & view need to be updated for this to work. In the mean time, if there's only one operator to choose from, it should be pre-selected.

**Comment by mbjones** [comment]
I agree with the spirit of this. It strikes me that these are the same operations as `>`, '<', and '==' for numeric values. Can we just use those rather than making a more complicated list?

**Comment by robyngit** [comment]
We could just re-use those operators (side note: technically it's `≥` and `≤`, see https://github.com/DataONEorg/collections-portals-schemas/issues/82). It's also very easy to use date-specific language for date filters, it's defined as a [property of the QueryRuleView](https://github.com/NCEAS/metacatui/blob/feature-query-builder/src/js/views/QueryRuleView.js#L239-L267)

**Comment by robyngit** [comment]
Related to this issue, when a definition dateFilter is built with only one value (just a `min`, `max`, or `value` element), it's not displayed correctly in the Query Builder. See Rule 2 in the randomly generated test portal, [destruct23](https://search-stage.test.dataone.org/edit/portals/destruct23)

**Comment by laurenwalker** [comment]
I agree that having an `is before` `is after` and `in the year` operators would be nice. However I think the slider we show now is good enough to achieve that result for people who need it (e.g. setting the year slider to `0 to 1990` would effectively be the same as choosing `is before 1990`), so this can be lower priority
